monitor/src/Monitor.cpp

```diff
@@ -0,0 +1,57 @@
+#include "Monitor.h"
+#include "UI.h"
+
+#include <iostream>
+#include <optional>
+#include <string>
+
+namespace Monitor {
+
+namespace {
+
+using Types::IdType;
+using Types::ValueType;
+using Types::Vital;
+using Types::Vitals;
+
+std::optional<Vital> findVital(const Vitals& vitals, const IdType id)
+{
+    if (const auto it = vitals.find(id); it != vitals.end())
+    {
+        return it->second;
+    }
+    return std::nullopt;
+}
+
+[[nodiscard]] UI::Status status(const Vital& vital, const ValueType value)
+{
+    const auto inRange = vital.range.contains(value);
+    const auto message = inRange ? "OK" : vital.errorMessage;
+    return UI::Status{.ok = inRange, .message = message};
+}
+
+void report(const Vitals& vitals, const UI::Readings& readings, UI& ui)
+{
+    for (const auto& [id, value] : readings)
```

Comment:
Good strategy of iterating over the vitals. Addition of a vital would not need this function to change.
In that way, this function is 'closed for modification', which means it doesn't need to be re-tested when we add new parameters.